pkg/controller/operconfig/migration.go

```diff
@@ -32,12 +45,99 @@ func migrateEgressFirewallCRs(ctx context.Context, operConfig *operv1.Network, c
 	return nil
 }
 
+func enableMulticastOVN(ctx context.Context, client cnoclient.Client) error {
+	// 1. query for netnamespaces
+	netNamespaceList, err := cnoclient.ListAllOfSpecifiedType(gvrNetnamespace, ctx, client)
+	if err != nil {
+		return err
+	}
+
+	// 2. iterate through netnamespaces
+	//    - any with multicast-enabled="true" annotation will cause an update to the corresponding
+	//      namespace to add the necessary OVN annotation
+	for _, nns := range netNamespaceList {
+		if nns.Object["metadata"].(map[string]interface{})["annotations"] == nil {
+			continue
+		}
+		multicastAnnotation := nns.GetAnnotations()["netnamespace.network.openshift.io/multicast-enabled"]
+		if multicastAnnotation == "true" {
+			// update namespace to have the same annotation
+			nspStr := fmt.Sprintf("%v", nns.Object["netname"])
+			namespaceObj, err := client.Default().Kubernetes().CoreV1().Namespaces().Get(ctx, nspStr, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			if namespaceObj.Annotations == nil {
+				namespaceObj.Annotations = make(map[string]string)
+			}
+			namespaceObj.Annotations["k8s.ovn.org/multicast-enabled"] = "true"
+
+			if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+				_, err := client.Default().Kubernetes().CoreV1().Namespaces().Update(ctx, namespaceObj, metav1.UpdateOptions{})
+				return err
+			}); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func enableMulticastSDN(ctx context.Context, client cnoclient.Client) error {
+	multicastRollbackReady, err := netNamespacesExistForAllNamespaces(ctx, client)
+	if !multicastRollbackReady {
+		return nil // wait for all SDN netnamespace resources to be created before rollback
+	} else if err != nil {
+		return err
+	}
+
+	// 1. query for namespaces
+	namespaceList, err := cnoclient.ListAllNamespaces(ctx, client)
+	if err != nil {
+		return err
+	}
+
+	// 2. iterate through namespaces
+	//    - any with multicast-enabled=true annotation will cause an update to the corresponding
+	//      netnamespace to add the necessary SDN annotation
+	for _, ns := range namespaceList {
+		if ns.Annotations["k8s.ovn.org/multicast-enabled"] == "true" {
+			// update netnamespace to have multicast annotation
+			netNamespaceObj := &uns.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "network.openshift.io/v1",
+					"kind":       "NetNamespace",
+					"metadata": map[string]interface{}{
+						"annotations": map[string]interface{}{
+							"netnamespace.network.openshift.io/multicast-enabled": "true",
```

Comment:
It would be better to use constants to store the annotation keys we use in this file.

Reply:
Yes... I was going to say it should get the ovn annotation names by vendoring the ovn-kubernetes sources and getting them from there, but it's not even exported in ovn-kubernetes...

But having them as constants at least here (rather than hardcoding them in multiple places) is definitely good.